infra/src/Pos/Infra/Communication/Types/Relay.hs

```diff
@@ -126,4 +134,44 @@ data RelayLogEvent =
     | EnqueueDequeueTime !Integer
     deriving Show
 
+instance Bi (DataMsg MCCommitment) where
+    encode (DataMsg (MCCommitment signedComm)) = encode signedComm
+    decode = DataMsg . MCCommitment <$> decode
+
+instance Bi (DataMsg MCOpening) where
+    encode (DataMsg (MCOpening sId opening)) = encodeListLen 2 <> encode sId <> encode opening
+    decode = do
+        enforceSize "DataMsg MCOpening" 2
+        DataMsg <$> (MCOpening <$> decode <*> decode)
+
+instance Bi (DataMsg MCShares) where
+    encode (DataMsg (MCShares sId innerMap)) = encodeListLen 2 <> encode sId <> encode innerMap
+    decode = do
+        enforceSize "DataMsg MCShares" 2
+        DataMsg <$> (MCShares <$> decode <*> decode)
+
+instance Bi (DataMsg MCVssCertificate) where
+    encode (DataMsg (MCVssCertificate vss)) = encode vss
+    decode = DataMsg . MCVssCertificate <$> decode
```

Comment:
While these types are defined below in `sinbin`, the `DataMsg` wrapper type is defined in this file, so the instances must live here too.